scalyr_agent/agent_main.py

```diff
@@ -757,40 +757,52 @@ def __detailed_status(
             return 1
 
         return_code = 0
-        fp = open(status_file)
-        for line in fp:
-            if not health_check:
-                print(line.rstrip())
-
-            if status_format == "json" or health_check:
-                health_result = self.__find_health_result_in_status_json(line)
-                if health_result:
-                    if health_check:
-                        print("Health check: %s" % health_result)
-                    if health_result != "Good":
-                        return_code = 2
-                elif health_check:
-                    print("Cannot get health check result.")
-            elif (
-                status_format == "text"
-                and "Health check:" in line
-                and not re.match(r"^Health check\:\s+Good$", line.strip())
-            ):
-                return_code = 2
-        fp.close()
+
+        with open(status_file, "r") as fp:
+            content = fp.read()
+
+        if not health_check:
+            # Regular non-health check invocation, just print the stats
+            print(content)
+            return return_code
+
+        # Health check invocation, try to parse status from the report and print and handle it here
+        health_result = self.__find_health_result_in_status_json(content)
+
+        if health_result != "Good":
```

Comment:
Do we need this check here considering we have `if health_result in ["good", "bad"]:` right below it? Also since `__find_health_result_in_status_json` returns `.lower()`ed strings can we ever hit this?

Reply:
I pushed more refactoring so that code is now out of update, please have a look again.